### PR TITLE
Feature - 2765 -  Replace skills filters

### LINKED ISSUE #2765: Replace CMO skills filter in request page.
[Wireframe ](https://www.figma.com/proto/XS4Ag6GWcgdq2dBlLzBkay/Admin-v3?page-id=0%3A1&node-id=949%3A5635&viewport=-1637%2C-2263%2C0.73&scaling=min-zoom&starting-point-node-id=949%3A5635&show-proto-sidebar=1)

### **OLD SKILLS**

This change should be made to the NEW SearchForm.

image

### **NEW SKILLS**

image

### PR DESCRIPTION
**Pull request by esizer**
Resolves #2765 

## Summary

This replaces the CMO Assets checkboxes with Skills on the `/talent/search` page.

## Testing

1. Navigate to `/talent/search`
2. Scroll down to "Skills as filters"
3. Add a few skills
4. Ensure a query occurs and proper results are returned

### CODE REVIEW

File: frontend/talentsearch/src/js/components/search/SearchForm.tsx [hunk 1]
```diff
@@ -40,6 +41,7 @@ export type FormValues = Pick<
 > & {
   languageAbility: LanguageAbility | typeof NullSelection;
   classifications: string[] | undefined;
+  skills: string[] | undefined;
   cmoAssets: string[] | undefined; // TODO REMOVE WHEN REPLACING CMOASSETS
```
Comment (vd1992) [inline frontend/talentsearch/src/js/components/search/SearchForm.tsx:45]: ah, there are some remnants of CMOassets to clean up
I think I marked some with todo's but might need to ctrl+F to be safe

**Comment by petertgiles** [comment]
This is looking really great, Eric.  However, in my testing I found that my selected skills didn't carry over to the create request page.  Is this happening for you?

https://user-images.githubusercontent.com/8978655/175960491-59e18404-1437-4655-98c2-aed6168554e3.mp4

**Comment by esizer** [comment]
> This is looking better. The skills carry over to the request page now. 😄 I'm not seeing them show up on the request when I look it up in the admin site though. Do we even have a spot in the database to save those values?

I am actually not sure, but im guessing no. I assume this will need to be done before continuing and merging this?

**Comment by petertgiles** [comment]
> I assume this will need to be done before continuing and merging this?

Shoot, looks like we missed the dependency on #3106.  :cry:

**Comment by esizer** [comment]
> Shoot, looks like we missed the dependency on #3106. 😢

Oh no, good catch! Um, should this go back into draft/backlog and marked as blocked?

**Comment by vd1992** [comment]
adjusting to the on-demand Chromatic, I think after resolving merge conflicts it might be a good time to run it once
would be handy for reviewing

**Comment by vd1992** [comment]
the deprecated CMO assets chain appears a bit off currently, the candidate count is right but it doesn't show the selected skill
set APPLICANTSEARCH to false -> php rebuild

image

then hit Request Candidates

image

**Comment by esizer** [comment]
> the deprecated CMO assets chain appears a bit off currently, the candidate count is right but it doesn't show the selected skill

Fixed with:[fix deprecated request filters](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3170/commits/77e0534d290f97e30260a9e91aaef368ea7608b9)

**Review by vd1992 — APPROVED** [review]
- things appear to work 
- Chromatic diff seems non-concerning (faker shift + tweaking a component to be more reusable)
- some hooks floating around for the update `/Request` page issue, that will be a more thorough issue/review

Nice work!

**Pull request merged**